### PR TITLE
versioning fixes and updates to appveyor config

### PR DESCRIPTION
**Pull request by amarekano**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -117,9 +117,17 @@ def get_package_data():
 				data[""].append(os.path.join(root, filename)[11:])
 	return data
 
+def get_version():
+	version_cmd = ['git', 'describe', '--tags', '|' ,'cut', '-c', '1-5']
+
+	if platform == "linux2" or platform == "linux":
+		return subprocess.check_output(' '.join(version_cmd),shell=True)
+	elif platform == "win32":
```
Comment (HenryHoggard) [inline setup.py]: Command doesn't work on windows.

File: setup.py [hunk 2]
```diff
@@ -117,9 +117,17 @@ def get_package_data():
...
+		return subprocess.check_output(version_cmd,shell=True)
+
 setuptools.setup(
   name = meta.name,
-  version = str(meta.version),
+  version = get_version(),
```
Comment (HenryHoggard) [inline setup.py]: No benefit over hardcoded version unless tired in with build process.

File: .appveyor.yml [hunk 3]
```diff
@@ -1,8 +1,6 @@
 environment:
     
   matrix:
-    - PYTHON: "C:\\Python27"
-      JAVA_HOME: "C:\\Program Files (x86)\\Java\\jdk1.7.0"
     - PYTHON: "C:\\Python27-x64"
```
Comment (HenryHoggard) [inline .appveyor.yml]: 32 bit support is required.

File: setup.py [hunk 4]
```diff
@@ -117,9 +117,17 @@ def get_package_data():
 				data[""].append(os.path.join(root, filename)[11:])
 	return data
 
+def get_version():
+	version_cmd = ['git', 'describe', '--tags', '|' ,'cut', '-c', '1-5']
```
Comment (HenryHoggard) [inline setup.py]: doesn't work if downloaded using release zip/tar.gz

File: setup.py (hunk 1 above)
Comment (amarekano) [reply setup.py]: Which bits don't work on windows?

File: setup.py (hunk 4 above)
Comment (amarekano) [reply setup.py]: Yes unfortunately. I guess we're going to have to default to manual versioning for now.

**Pull request merged**